src/modules/DNS.rb

```diff
@@ -282,8 +282,8 @@ def ProposeHostname
     def Read
       return true if @initialized == true
 
-      @dhcp_hostname = dhclient_set_hostname
-      @write_hostname = get_write_hostname_to_hosts
+      @dhcp_hostname = dhclient_set_hostname || false
```

Comment:
np: i'm thinking of adding a note that "no" is default value in sysconfig in both cases.